libdevcrypto/Common.cpp

```diff
@@ -138,10 +160,50 @@ bool dev::decryptECIES(Secret const& _k, bytesConstRef _cipher, bytes& o_plainte
 
 bool dev::decryptECIES(Secret const& _k, bytesConstRef _sharedMacData, bytesConstRef _cipher, bytes& o_plaintext)
 {
-	bytes io = _cipher.toBytes();
-	if (!Secp256k1PP::get()->decryptECIES(_k, _sharedMacData, io))
+	// interop w/go ecies implementation
+
+	// io_cipher[0] must be 2, 3, or 4, else invalidpublickey
+	if (_cipher.empty() || _cipher[0] < 2 || _cipher[0] > 4)
```

Comment:
Probably better to have a class for SEC.1 key (mentioned above) and just use validations on that class here.